libs/csrf/csrfprotector.php

```diff
@@ -396,33 +396,27 @@ public static function ob_handler($buffer, $flags)
 		    $buffer = preg_replace("/<body[^>]*>/", "$0 <noscript>" .self::$config['disabledJavascriptMessage'] .
 		    	"</noscript>", $buffer);
 
-		    $arrayStr = '';
+		    $urls = array();
 		    if (!self::useCachedVersion()) {
 		    	try {
 		    		self::createNewJsCache();
 		    	} catch (exception $ex) {
 		    		if (self::$config['verifyGetFor']) {
-						foreach (self::$config['verifyGetFor'] as $key => $value) {
-							if ($key != 0) $arrayStr .= ',';
-							$arrayStr .= "'". $value ."'";
-						}
+						$urls = self::$config['verifyGetFor'];
 					}
 		    	}
 		    }
 
-		    $script = '<script type="text/javascript" src="' .self::$config['jsUrl']
-		    	.'"></script>' .PHP_EOL;
-
-		    $script .= '<script type="text/javascript">' .PHP_EOL;
-		    if ($arrayStr !== '') {
-		    	$script .= 'CSRFP.checkForUrls = [' .$arrayStr .'];' .PHP_EOL;
-		    }
-		    $script .= 'window.onload = function() {' .PHP_EOL;
-		    $script .= '	csrfprotector_init();' .PHP_EOL;
-		    $script .= '};' .PHP_EOL;
-		    $script .= '</script>' .PHP_EOL;
+		    //implant hidden fields with check url information for reading in javascript
```

Comment:
Normally when the config file is changed the main js file is automatically changed, that's what `self::createNewJsCache();` LINE#202 does. Now if due to file permission error it fails, we had to insert the urls inline so that the system doesn't fail.
So insert this hidden field only when say `count($urls) > 0`. Also JS should try to retrieve urls from the hidden field only when the hidden field is available.
Initiating the `init()` on `onload` event is good, and we need to be compliant with CSP rules.